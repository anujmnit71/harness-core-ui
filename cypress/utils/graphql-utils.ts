export const hasOperationName: (req: any, operationName: string) => boolean = (req, operationName) => {
  const { body } = req
  // eslint-disable-next-line no-prototype-builtins
  return body.hasOwnProperty('operationName') && body.operationName === operationName
}

export const aliasQuery: (req: any, operationName: string) => void = (req, operationName) => {
  if (hasOperationName(req, operationName)) {
    req.alias = `gql${operationName}Query`
  }
}
