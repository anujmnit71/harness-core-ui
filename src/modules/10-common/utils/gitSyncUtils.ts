/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import type { GitSyncConfig } from 'services/cd-ng'
import { folderPathName, yamlFileExtension } from './StringUtils'

export const getRepoDetailsByIndentifier = (identifier: string | undefined, repos: GitSyncConfig[]) => {
  return repos.find((repo: GitSyncConfig) => repo.identifier === identifier)
}

export const validateFilePath = (filePath: string): boolean => {
  if (!filePath.endsWith(yamlFileExtension)) {
    return false
  } else {
    const fullPath = filePath.slice(0, filePath.length - yamlFileExtension.length)
    const subPaths = fullPath.split('/')
    return subPaths.every((folderName: string) => {
      return folderName.length && folderName.match(folderPathName)?.length
    })
  }
}
