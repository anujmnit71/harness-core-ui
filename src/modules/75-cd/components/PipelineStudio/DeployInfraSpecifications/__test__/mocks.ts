/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

export const envs = {
  status: 'SUCCESS',
  data: {
    totalPages: 1,
    totalItems: 1,
    pageItemCount: 1,
    pageSize: 100,
    content: [
      {
        environment: {
          accountId: 'zEaak-FLS425IEO7OLzMUg',
          orgIdentifier: 'default',
          projectIdentifier: 'Test_Yunus',
          identifier: 'dev',
          name: 'something',
          description: '',
          color: '#0063F7',
          type: 'Production',
          deleted: false,
          tags: {},
          version: 2
        },
        createdAt: 1640688198443,
        lastModifiedAt: 1640699193440
      }
    ],
    pageIndex: 0,
    empty: false
  },
  metaData: null,
  correlationId: '99ca7534-0816-4723-a892-da2463604b8f'
}
