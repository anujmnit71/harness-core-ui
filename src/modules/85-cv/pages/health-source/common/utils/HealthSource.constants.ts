/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

export const HealthSourceFieldNames = {
  MANUAL_QUERY: 'isManualQuery',
  IDENTIFIER: 'identifier',
  METRIC_NAME: 'metricName',
  METRIC_DATA: 'metricData',
  GROUP_NAME: 'groupName',
  CONTINUOUS_VERIFICATION: 'continuousVerification',
  HEALTH_SCORE: 'healthScore',
  SLI: 'sli',
  RISK_CATEGORY: 'riskCategory',
  HIGHER_BASELINE_DEVIATION: 'higherBaselineDeviation',
  LOWER_BASELINE_DEVIATION: 'lowerBaselineDeviation'
}
