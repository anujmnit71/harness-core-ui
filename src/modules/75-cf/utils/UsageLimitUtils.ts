/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import type { ReactNode } from 'react'
import { BannerType } from '@common/layouts/Constants'
import type { CheckFeatureReturn } from 'framework/featureStore/featureStoreUtil'
import { formatToCompactNumber } from '@cf/utils/CFUtils'
import type { UseStringsReturn } from 'framework/strings'

export const getBannerText = (
  getString: UseStringsReturn['getString'],
  monthlyActiveUsers: CheckFeatureReturn,
  additionalLicenseProps: Record<string, any>
): ReactNode => {
  const { isEnterpriseEdition, isFreeEdition, isTeamEdition } = additionalLicenseProps

  const clientMauUsageCount = Number(monthlyActiveUsers?.featureDetail?.count)
  const clientMauPlanLimit = Number(monthlyActiveUsers?.featureDetail?.limit)

  const clientMauUsagePercentage = Math.trunc((clientMauUsageCount / clientMauPlanLimit) * 100)
  const clientMauPlanLimitFormatted = formatToCompactNumber(clientMauPlanLimit)

  const showInfoBanner = clientMauPlanLimit > 0 && clientMauUsagePercentage >= 90 && clientMauUsagePercentage < 100
  const showWarningBanner = clientMauPlanLimit > 0 && clientMauUsagePercentage >= 100

  let message!: string
  let bannerType!: BannerType

  if (isFreeEdition) {
    if (!showInfoBanner) {
      return {
        message: () =>
          getString('cf.planEnforcement.freePlan.approachingLimit', {
            clientMauUsagePercentage,
            clientMauPlanLimitFormatted
          }),
        bannerType: BannerType.INFO
      }
    }

    if (showWarningBanner) {
      return {
        message: getString('cf.planEnforcement.freePlan.upgradeRequired', {
          clientMauPlanLimitFormatted
        }),
        bannerType: BannerType.LEVEL_UP
      }
    }
  }

  if (isTeamEdition || isEnterpriseEdition) {
    if (showInfoBanner) {
      return {
        message: () =>
          getString('cf.planEnforcement.teamEnterprisePlan.approachingLimit', {
            clientMauUsagePercentage
          }),
        bannerType: BannerType.INFO
      }
    }

    if (showWarningBanner) {
      return {
        message: () => getString('cf.planEnforcement.teamEnterprisePlan.upgradeRequired'),
        bannerType: BannerType.LEVEL_UP
      }
    }
  }

  return {
    message,
    bannerType
  }
}
