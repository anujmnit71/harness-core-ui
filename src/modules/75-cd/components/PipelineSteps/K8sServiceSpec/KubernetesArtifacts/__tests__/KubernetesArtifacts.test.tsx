/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { render } from '@testing-library/react'

import { MultiTypeInputType } from '@harness/uicore'
import { TestWrapper } from '@common/utils/testUtils'

import { ArtifactSourceBaseFactory } from '@cd/factory/ArtifactSourceFactory/ArtifactSourceBaseFactory'
import type { ArtifactListConfig, ServiceSpec } from 'services/cd-ng'
import { StepViewType } from '@pipeline/components/AbstractSteps/Step'
import { KubernetesArtifacts } from '../KubernetesArtifacts'
import { artifacts, template } from '../KubernetesPrimaryArtifacts/__tests__/mocks'

describe('<KubernetesArtifacts /> tests', () => {
  test('snapshot test for kubernetes artifacts', () => {
    const { container } = render(
      <TestWrapper>
        <KubernetesArtifacts
          type={'artifact'}
          initialValues={{}}
          template={template as ServiceSpec}
          artifacts={artifacts as ArtifactListConfig}
          readonly
          stageIdentifier="stage-0"
          artifactSourceBaseFactory={new ArtifactSourceBaseFactory()}
          allowableTypes={[MultiTypeInputType.FIXED, MultiTypeInputType.RUNTIME, MultiTypeInputType.EXPRESSION]}
          fromTrigger={false}
          stepViewType={StepViewType.DeploymentForm}
        />
      </TestWrapper>
    )
    expect(container).toMatchSnapshot()
  })
})
