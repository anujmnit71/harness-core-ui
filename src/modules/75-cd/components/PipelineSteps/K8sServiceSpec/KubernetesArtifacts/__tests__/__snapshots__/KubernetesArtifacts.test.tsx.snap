// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<KubernetesArtifacts /> tests snapshot test for kubernetes artifacts 1`] = `
<div>
  <div
    class="nopadLeft accordionSummary"
    id="Stage.stage-0.Service.Artifacts.Sidecars"
  />
</div>
`;
